report_xlsx/models/ir_report.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2015 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class ReportAction(models.Model):
+    _inherit = 'ir.actions.report'
+
+    report_type = fields.Selection(selection_add=[("xlsx", "XLSX")])
+
+    @api.model
+    def render_xlsx(self, docids, data):
+        report_model_name = 'report.%s' % self.report_name
+        report_model = self.env.get(report_model_name)
+        if report_model is None:
+            raise UserError(_('%s model was not found' % report_model_name))
+        return report_model.with_context({
+            'active_model': self.model
```

Comment:
@ovnicraft, please, can yo add this code:
```suggestion
            'active_model': self.model,
            'lang': self.env.user.partner_id.lang
```
because, translation report is wrong. Need pass language by context. Thanks.

Reply:
@RickHunterEc I am less than convinced by this.  lang is usually already set in context in UI or the caller may have specifically set context lang to something other than user lang (e.g. partner.lang in an email attachment). I believe the correct way to do this is 

`report_model.with_context(active_model=self.model)`

to stop the existing context from being overwritten.

Reply:
Yes, language its always on context.
Force the lang on user could break the normal use of this.

Reply:
@ovnicraft You will still need to make the change.

`with_context(dict())` sets a new context effectively removing lang from the dict..
`with_context(**kwargs)` updates current context so either pass the dict with ** or make the args active_model=self.model etc

Reply:
> @RickHunterEc I am less than convinced by this. lang is usually already set in context in UI or the caller may have specifically set context lang to something other than user lang (e.g. partner.lang in an email attachment). I believe the correct way to do this is
> 
> `report_model.with_context(active_model=self.model)`

This is already done.
> 
> to stop the existing context from being overwritten.

Reply:
weird, its still showing the old code when I look at the file.

Reply:
Check it https://github.com/ovnicraft/reporting-engine/blob/12.0-mig-report_xlsx/report_xlsx/models/ir_report.py#L20

Reply:
@ovnicraft It still shows the same using the dict constructor method instead of kwargs.  The dict constructor overwrites the entire context, keyword args updates the existing.  So using dict constructor as shown there, you effectively remove any context lang from the dict.

I had the same issue on base_report_to_printer.

Reply:
@ovnicraft  and @gdgellatly , I agree with you.

Reply:
I fix it on this weekend

Reply:
Hello, is there any update about this task?

Reply:
I recently push a commit with `dict()` as @gdgellatly recommended, BTW i am sure if this change fix problem so i am not passing active model, maybe i need call `with_context(active_model=self.model)` after `dict()` constructor call